### PR TITLE
Feature/axon kotlin test

### LINKED ISSUE #67: Improve AggregateTextFixture
### Scenario

* extension-kotlin version: 0.1.0
* Axon Framework version: 4.4.x
* Description of your use case: (detailed description or executable reproducer, e.g. GitHub repo)

### Current Behaviour
1.  `AggregateTestFixture` currently requires a `Class<T> aggregateType` to be created. 
2. The method named `when` of `AggregateTestFixture` is used to set up the test, which forces Kotlin users to escape the method name since `when` is a keyword in Kotlin

### Wanted Behaviour

1. instead of passing Java's class object to `AggregateTestFixture`, a factory method with a reified generic parameter can be used instead. That would allow for a syntax like zaggregateTestFixture<GiftCard>()` instead of `AggregateTestFixture(GiftCard.javaClass)`
2. An extension method on `AggregateTestFixture` which will alias/delegate to `when`, but will be called something else that is not a keyword. One option could be `whenever`, but anything else that make sense will do.
3. This will require a new `test` module of the extension, to allow extension users to use the module only for testing.

### PR DESCRIPTION
**Pull request by zambrovski**
Will fix #67

### CODE REVIEW

**Review by sandjelkovic — CHANGES_REQUESTED** [review]
The package name should be changed to indicate that these are test utilities and differentiate from the regular extension methods. Appending `.test` to the package should be enough, as there's no need to split aggregate and saga utilities for now. 
We can always segregate it in the future if there's the need for it. 

Otherwise, looks good to me 👍

**Review by sandjelkovic — APPROVED** [review]

**Review by smcvb — APPROVED** [review]
Looks good to me :+1:

**Review by Arnaud-J — COMMENTED** [review]
Hi!

After trying this code locally for my project, I had trouble using the defined functions. **I am still very new to Kotlin, therefore I might be missing something**.  
Nonetheless, here are some comments on how I would refactor this code. Please tell me if I am on the wrong path.

File: kotlin-test/src/main/kotlin/org/axonframework/extension/kotlin/test/FixtureExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.axonframework.test.saga.SagaTestFixture
+import kotlin.reflect.KClass
+
+/**
+ * Creates an aggregate test fixture for aggregate [T].
+ * @param T reified type of the aggregate.
+ * @return aggregate test fixture.
+ * @since 0.2.0
+ */
+inline fun <reified T : Any> AggregateTestFixture<T>.aggregateTestFixture() =
```
[the remark this answers is not in the document]
Comment (sandjelkovic) [reply kotlin-test/src/main/kotlin/org/axonframework/extension/kotlin/test/FixtureExtensions.kt:31]: Hey Arnaud,

Thank you for noticing this! You are completely correct, these are supposed to be factory methods for fixtures (Aggregate and Saga fixtures)

I've created an issue to handle this, you can find it here https://github.com/AxonFramework/extension-kotlin/issues/92

If you are up for it, a PR would be most welcome

[the remark this answers is not in the document]
Comment (Arnaud-J) [reply kotlin-test/src/main/kotlin/org/axonframework/extension/kotlin/test/FixtureExtensions.kt:31]: Associated PR: https://github.com/AxonFramework/extension-kotlin/pull/93

File: kotlin-test/src/main/kotlin/org/axonframework/extension/kotlin/test/FixtureExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        AggregateTestFixture(T::class.java)
+
+/**
+ * Alias for the `when` method to avoid name clash with Kotlin's `when`.
+ * @param command command to pass to the when method.
+ * @param T aggregate type.
+ * @return result validator.
+ * @since 0.2.0
+ */
+fun <T : Any> AggregateTestFixture<T>.whenever(command: Any): ResultValidator<T> = this.`when`(command)
```
[the remark this answers is not in the document]
Comment (sandjelkovic) [reply kotlin-test/src/main/kotlin/org/axonframework/extension/kotlin/test/FixtureExtensions.kt:41]: Hey Arnaud, 

Yes, this change also makes perfect sense. Just please make sure both Aggregate and Saga fixtures are covered, both in cleaning up and example (test) usage, same as for aggregate factories.

**Pull request merged**